engine/src/test/java/org/camunda/bpm/engine/test/api/context/DelegateExecutionContextTest.java

```diff
@@ -42,70 +47,197 @@
  */
 public class DelegateExecutionContextTest {
 
-  protected static final BpmnModelInstance DELEGATION_PROCESS = Bpmn.createExecutableProcess("process1")
+  protected static final BpmnModelInstance SERVICE_TASK_DELEGATE_PROCESS = Bpmn.createExecutableProcess()
+      .startEvent()
+      .serviceTask("serviceTask1")
+        .camundaClass(DelegateClass.class.getName())
+      .endEvent()
+      .done();
+
+
+  protected static final BpmnModelInstance EXEUCTION_LISTENER_PROCESS = Bpmn.createExecutableProcess()
+      .startEvent()
+        .camundaExecutionListenerClass(ExecutionListener.EVENTNAME_START, ExecutionListenerImpl.class.getName())
+      .endEvent()
+      .done();
+
+  protected static final BpmnModelInstance SIGNAL_EVENT_PROCESS = Bpmn.createExecutableProcess()
+      .startEvent()
+        .intermediateCatchEvent("catchSignal")
+          .signal("${delegateExecutionContextBean.getCurrentActivityId()}")
+      .endEvent()
+      .done();
+
+  protected static final BpmnModelInstance ONE_TASK_PROCESS_SUBPROCESS_SIGNAL_INTERMEDIATE_EVENT = Bpmn.createExecutableProcess()
+      .startEvent()
+        .userTask("A")
+      .endEvent()
+      .subProcess("subprocess")
+        .embeddedSubProcess()
           .startEvent()
-          .serviceTask("serviceTask1")
-            .camundaClass(DelegateClass.class.getName())
+          .intermediateCatchEvent("catchSignal")
```

Comment:
This case should rather test an event subprocess with a signal start event inside the embedded subprocess. 

What it currently tests is the same logic as in `shouldFindSignalActivityIdThroughDelegateExecutionContext_IntermediateSignalCatchEvent`: It creates the event subscription as they are created in `PvmAtomicOperationCreateScope`.

What we want to test with this case: That the logic that creates scope executions in `PvmAtomicOperationActivityInitStack` creates the event subscriptions correctly, too. This atomic operations is only responsible for creating the scope executions within which the new execution starts executing (i.e. the embedded subprocess in this test case). When the embedded subprocess execution is created, it will create the subscription for any event subprocess contained. This will then happen as part of `PvmAtomicOperationActivityInitStack`.